scripts/bfasst/compare_waveforms/file_parsing/parse_diff.py

```diff
@@ -1,18 +1,23 @@
 """A module used to accurately check the difference between VCD files"""
 
 
-def parse_signals(line):
+def init_data():
 
-    """Is used in the header of a VCD file to find the names of every signal, it's corresponding
-    symbol, and it's bitsize."""
+    """Sets all of the initial variables for data"""
 
-    words = []
-    new_word = False
-    if "$var wire" in line:
-        words = parse_io(line, "wire", words, new_word)
-    if "$var reg" in line:
-        words = parse_io(line, "reg", words, new_word)
-    return words
+    data = {}
+    data["time"] = "0"
+    data["name"] = []
+    data["binary_name"] = []
+    data["signal"] = []
+    data["binary_signal"] = []
+    data["state"] = []
+    data["binary_state"] = []
+    data["returned_signal"] = []
+    data["past_header"] = False
+    data["past_definitions"] = False
+
+    return data
 
 
 def parse_io(line, input_output, words, new_word):
```

Comment:
Just fyi, as long as your code works and follows the coding standard, I'll approve it. However, this function in particular is very amenable to some of the documented parser techniques I listed in the examples section of the coding standard.  I don't know if you've taken cs 236, but those techniques are exactly for this purpose.  You also might look at python's `split` function, which looks like it would greatly reduce your logic.